### ISSUE TITLE
Fix test run and setup GitHub Action for PRs

### ISSUE THREAD

**Issue by indigoxela** [backdrop-contrib/b#117]
Not sure yet, if this is a bigger problem, but the Travis checks seem to be broken after the major code refactoring in #5.

**Comment by ghost** [comment]
Yep, I can imagine. Updated code, haven't touched tests yet...

Thanks for the issue.

**Comment by indigoxela** [comment]
> Thanks for the issue.

You don't seem to make use of my issues, though... See https://github.com/backdrop-contrib/b/issues/118, which is a duplicate of this one IMO. :wink: 

Close either of it, I'll leave it up to you.

**Comment by ghost** [comment]
Oops, sorry @indigoxela, forgot about this one 🙂 Indeed my issue is a duplicate.

Here's what I posted in https://github.com/backdrop-contrib/b/issues/118:

> Since the recent re-write of the code, we need to re-write tests too, and add tests for any new functionality not previously covered.
>
> Also see if Travis is still the best way to run tests. Maybe see if GitHub actions is easier/better...?

**Comment by indigoxela** [comment]
Hm. It's not only about fixing tests alone, there's also broken functionality. @BWPanda Should I open an extra issue for all broken things I find, or do you prefer a "meta issue"?

**Comment by ghost** [comment]
@indigoxela If you've found bugs, please open issues for them (bugs relating to the same command can go into the same issue, otherwise separate issues is probably better). Thanks.

**Comment by ghost** [comment]
> > Also see if Travis is still the best way to run tests. Maybe see if GitHub actions is easier/better...?

Note to self: Read through https://www.nymedia.no/en/blog/why-we-decided-migrate-from-travis-to-github-action to see if says anything interesting/relevent on the subject...

**Comment by indigoxela** [comment]
After #122 it seems that many (most?) of the failing tests are just searching for the wrong string pattern (quoting character changed).

Additionally, phpcs nags a lot, but partly because the profile in use and the actual Backdrop standards differ. There *are* some coding standard violations, though.

**Comment by ghost** [comment]
Since we're trying to setup our testing environment in Lando (https://github.com/backdrop-contrib/b/issues/72), then it'd be nice if we can re-use that environment for PR tests. i.e. let's explore using Lando in a GitHub Action, instead of setting up the regular/multisite installations from scratch. To that end, https://github.com/lando/hyperdrive might come in handy...

**Comment by indigoxela** [comment]
This former PR for i18n might be interesting re Lando: https://github.com/backdrop-contrib/i18n/pull/31/files.

I decided against Lando in that project for performance reasons, and also because personally I don't use Lando at all, but for your use-case it might be a good source for some copy-paste. :smiley:

**Comment by ghost** [comment]
Tests have been fixed and merged, now just need to work on getting the new automation happening for PRs using GitHub actions...

**Comment by ghost** [comment]
In case anyone's interested (and for posterity), tests need to run in a certain environment (i.e. they need a regular Backdrop install, plus a multisite install, and they need to know the filesystem paths, etc.). So I've heavily configured the included Lando file to setup this environment for tests. If you'd like to run tests yourself (for whatever reason, no judgement here), you'll have to use the Lando setup to do so (or else setup your own local environment to match). Once you've started Lando, just run `lando run-tests`!

**Comment by ghost** [comment]
Tests should now run automatically on any PR or push to the `1.x-1.x` branch.
